velox/common/memory/SharedArbitrator.cpp

```diff
@@ -296,15 +303,18 @@ uint64_t SharedArbitrator::shrinkCapacity(
     bool allowAbort) {
   const auto freeCapacityUpdateCb =
       folly::makeGuard([this]() { updateFreeCapacityMetrics(); });
-
-  ScopedArbitration scopedArbitration(this);
+  ArbitrationOperation op(targetBytes, pools);
+  op.needGlobalArbitration = true;
```

Comment:
Maybe put needGlobalArbitration to be in ctor as well, and default to false. This makes me think ArbitrationOpertion is more like a complex class than a struct. Maybe make ArbitrationOpertion a full fledged class with modifiers and accessors?